core/src/agent/state.rs

```diff
@@ -59,9 +128,45 @@ impl AgentState {
     pub fn top_chain_header(&self) -> Option<ChainHeader> {
         self.top_chain_header.clone()
     }
+
+    pub fn from_json_str(header_str: &str) -> serde_json::Result<Self> {
+        serde_json::from_str(header_str)
+    }
 }
 
-#[derive(Clone, Debug, PartialEq)]
+impl ToEntry for AgentState {
```

Comment:
This feels strange. The redux state is not an entry and we dont' want to commit redux states to source chains or share them in a DHT, right? Why do you want to treat the AgentState as an Entry?

Reply:
Just a copy and paste from addressable content. Will push this out.